scripts/core/CUDA.rst

```diff
@@ -0,0 +1,158 @@
+<%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+
+==========================
+CUDA UR Reference Document
+==========================
+
+This document gives general guidelines of how to use UR to load and build
+programs, and execute kernels on a CUDA device.
+
+Device code
+===========
+
+A CUDA device image may be made of PTX and/or SASS, two different kinds of
+device code for NVIDIA GPUs.
+
+CUDA device images can be generated by a CUDA-capable compiler toolchain. Most
+CUDA compiler toolchains are capable of generating PTX, SASS and/or bundles of
+PTX and SASS.
+
+PTX
+---
+
+PTX is a high level NVIDIA ISA which can be JIT compiled at runtime by the CUDA
+driver. In UR, this JIT compilation happens at ${x}ProgramBuild, where PTX is
+assembled into device specific SASS which then can run on device.
+
+PTX is forward compatible, so PTX generated for `.target sm_52` will be JIT
```

Comment:
Inline monospace in reStructuredText should be wrapped in double backticks not single backticks. I'll not make comments for ever instance in these documents but they should all be updated.

Reply:
D'oh thanks!